tests/unit/test_quantize.py

```diff
@@ -225,3 +226,68 @@ def test_quantize_with_empty_and_non_empty_files(tmpdir):
             )
             output_parquet_index_faiss = faiss.read_index(index_parquet_path)
             assert output_parquet_index_faiss.ntotal == len(expected_df)
+
+
+def test_index_correctness_in_distributed_mode(tmpdir):
+    min_size = 8000
+    max_size = 10240
+    dim = 512
+    nb_files = 5
+
+    # parquet
+    tmp_dir, _, _, expected_df, _ = build_test_collection_parquet(
+        tmpdir, min_size=min_size, max_size=max_size, dim=dim, nb_files=nb_files
+    )
+    temporary_indices_folder = os.path.join(tmpdir.strpath, "distributed_autofaiss_indices")
+    ids_path = os.path.join(tmpdir.strpath, "ids")
+    index, _ = build_index(
+        embeddings=tmp_dir,
+        distributed="pyspark",
+        file_format="parquet",
+        temporary_indices_folder=temporary_indices_folder,
+        max_index_memory_usage="100MB",
+        current_memory_available="120MB",
+        embedding_column_name="embedding",
+        index_key="IVF1,Flat",
+        should_be_memory_mappable=True,
+        metric_type="l2",
+        ids_path=ids_path,
+    )
+    query = faiss.rand((1, dim))
+    distances, ids = index.search(query, k=9)
+
+    ground_truth_index = faiss.index_factory(dim, "IVF1,Flat")
+    expected_array = np.stack(expected_df["embedding"])
+    ground_truth_index.train(expected_array)
+    ground_truth_index.add(expected_array)
+    ground_truth_distances, ground_truth_ids = ground_truth_index.search(query, k=9)
+
+    assert_array_equal(ids, ground_truth_ids)
+
+    # numpy
+    tmp_dir, _, _, expected_array, _ = build_test_collection_numpy(
+        tmpdir, min_size=min_size, max_size=max_size, dim=dim, nb_files=nb_files
+    )
+
+    index, _ = build_index(
+        embeddings=tmp_dir,
+        distributed="pyspark",
+        file_format="npy",
+        temporary_indices_folder=temporary_indices_folder,
+        max_index_memory_usage="100MB",
+        current_memory_available="120MB",
+        embedding_column_name="embedding",
+        index_key="IVF1,Flat",
+        should_be_memory_mappable=True,
+        metric_type="l2",
+    )
+
+    query = faiss.rand((1, dim))
+    distances, ids = index.search(query, k=9)
+
+    ground_truth_index = faiss.index_factory(dim, "IVF1,Flat")
+    ground_truth_index.train(expected_array)
+    ground_truth_index.add(expected_array)
+    ground_truth_distances, ground_truth_ids = ground_truth_index.search(query, k=9)
+
+    assert_array_equal(ids, ground_truth_ids)
```

Comment:
could you also check the mapping ids and not only the consecutive ids ?

Reply:
done now.